lib/matplotlib/widgets.py

```diff
@@ -209,7 +214,11 @@ def _motion(self, event):
         if not colors.same_color(c, self.ax.get_facecolor()):
             self.ax.set_facecolor(c)
             if self.drawon:
-                self.ax.figure.canvas.draw()
+                if self._useblit:
+                    self.ax.draw_artist(self.ax)
```

Comment:
Can you confirm that Button doesn't use copy_from_bbox/restore_region like the other two widgets because the button covers the entire axes anyways?

Reply:
Yes, and it's strictly a rectangle, so `copy_from_bbox`/`restore_region` would effectively be directly overwritten immediately.